### DISCUSSION TITLE
Foundry Raffle Compilation Error.

### DISCUSSION THREAD

**Discussion by Rayyan-Uddin** [Cyfrin/foundry-full-course-cu#2630] [Q&A] [answered]
Hey everyone! need assistance with solving this compilation issue,
![error1](https://github.com/user-attachments/assets/3824d2cf-efc3-4adb-8584-d173c691982c)
![error2](https://github.com/user-attachments/assets/dd346b65-91ae-458f-9f7a-708e2c62375a)
 Thanks.

**Comment by EngrPips** [comment]
Hello @Rayyan-Uddin, It will be more helpful if you copy the error and paste it here with nice formatting, as I can see the information in the image you sent

**Reply by Rayyan-Uddin** [reply]
ok sure.

Compiler run failed:
Error (3656): Contract "Raffle" should be marked as abstract.
  --> src/raffle.sol:34:1:
   |
34 | contract Raffle is VRFConsumerBaseV2Plus {
   | ^ (Relevant source part starts here and spans across multiple lines).
Note: Missing implementation: 
   --> lib/chainlink-brownie-contracts/contracts/src/v0.8/vrf/dev/VRFConsumerBaseV2Plus.sol:135:3:
    |
135 |   function fulfillRandomWords(uint256 requestId, uint256[] calldata randomWords) internal virtual;
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (4266): Uninitialized "constant" variable.
  --> src/raffle.sol:47:5:
   |
47 |     uint32 private constant NUM_WORD;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (7407): Type bytes32 is not implicitly convertible to expected type uint256.
  --> src/raffle.sol:61:21:
   |
61 |         i_keyHash = gasLane;
   |                     ^^^^^^^

Error (9553): Invalid type for argument in function call. Invalid implicit conversion from uint256 to bytes32 requested.
  --> src/raffle.sol:88:26:
   |
88 |                 keyHash: i_keyHash,
   |                          ^^^^^^^^^

**Reply by EngrPips** [reply, marked as answer]
Hello @Rayyan-Uddin, There is a lot of stuff wrong in your code at a time, so I think it is best that you compare your code with the official code of the `lottery section`. You can find the code here https://github.com/Cyfrin/foundry-smart-contract-lottery-cu

**Reply by Rayyan-Uddin** [reply]
Okay.
